src/main/java/use_cases/add_edit_expenses_use_case/ExpenseUCI.java

```diff
@@ -131,76 +129,75 @@ public ExpenseOD editExpenseInMonth(ExpenseID expenseIDEdit) throws EntityExcept
                 if (expenseID.getIsRecurringExpense()) {
-                    if (checkHaveSameNameInList(recurringExpenseList)) {
+                    //Repeated recurring expense name fail: When a user tries to edit the current expense to a recurring expense but the name already exist as a recurring expense.
+                    if (checkHaveSameNameInList(recurringExpenseList,expenseID.getName())) {
                         ExpenseOD expenseODFailEdit = new ExpenseOD("There is a recurring expense with this name, you don't need to add recurring expense in month! " +
                                 "(If this is not the same expense, please use another name!)");
                         return expenseOB.fail(expenseODFailEdit);
                     }else {
                         selectedExpense.setName(expenseIDEdit.getName());
                         expenseIDEdit.setValue(valueDouble);
                         selectedExpense.setValue(valueDouble);
-                        session.deleteRecurExpense(expenseID.getName());
+                        session.addRecurExpense(selectedExpense);
                         selectedExpense.setCategory(selectedCategory);
-                        ExpenseOD expenseODSuccessEditRecurring = new ExpenseOD("You have updated all changes of this expense to the category selected in current month!");
-                        return expenseOB.success(expenseODSuccessEditRecurring);
-                    }
-                }else {
+                        // Success edit to new recurring expense.
+                        ExpenseOD expenseODSuccessEditRecurring = new ExpenseOD("You have updated all changes of this new recurring expense!");
+                        return expenseOB.success(expenseODSuccessEditRecurring);}
+                }else if(checkHaveSameNameInList(recurringExpenseList, expenseID.getOldExpense())&& !expenseIDEdit.getIsRecurringExpense()) {
                     selectedExpense.setName(expenseIDEdit.getName());
                     expenseIDEdit.setValue(valueDouble);
                     selectedExpense.setValue(valueDouble);
-                    session.addRecurExpense(selectedExpense);
-                    selectedExpense.setCategory(findCategory(categoryList, "Other"));
-                    ExpenseOD expenseODSuccessEditRecurring = new ExpenseOD("You have updated all changes of this expense to a recurring expense in current session!");
+                    session.deleteRecurExpense(expenseID.getName());
+                    selectedExpense.setCategory(selectedCategory);
+                    //Success edit to remove recurring expense.
+                    ExpenseOD expenseODSuccessEditRecurring = new ExpenseOD("You have updated all changes of this expense and it is no longer a recurring expense!");
                     return expenseOB.success(expenseODSuccessEditRecurring);
                 }
-                    }
 
             selectedExpense.setName(expenseIDEdit.getName());
```

Comment:
For clarity, everything above the catch should be in an else clause.

Reply:
Also, notice how you have the same three lines of code repeated wherever you are editing an expense? You could make an editExpense() private helper method with those three set_ lines and that would make the code up here cleaner!

Reply:
That sounds like a good idea

Reply:
Thanks! :D you could probably do a similar thing for adding as well!

Reply:
I looked at it for addExpense, but it will just be two lines of code and it will have to pass in the same amount of parameters. So I think I'll just leave it as is! (Also one of them will be MonthlyStorage.addeEpense() which is going to throw an entityException and I will like to avoid it at all costs lol)

Reply:
Sounds good!! Omg real the EntityException there has gotten so annoying so yeah best to leave it out!